### PR TITLE
fix(proxy-rewrite): query string discarded when use_real_request_uri_…

### LINKED ISSUE #12623: bug: proxy-rewrite discard the query string when use_real_request_uri_unsafe is true
### Current Behavior

Using the proxy-rewrite plugin to modify the request uri, if `use_real_request_uri_unsafe` is set to `true`, the queryS tring of original request will be discarded. And returned to normal when `use_real_request_uri_unsafe`  is set to `false`(default), 

### Expected Behavior

Whether or not `use_real_request_uri_unsafe` is enabled, the queryString should not be discarded.

### Error Logs

_No response_

### Steps to Reproduce

1. Add route with proxy-rewrite plugin:
```
curl -XPUT -H "x-api-key:${ADMIN_KEY}" http://localhost:9180/admin/apisix/routes \
-d '{
    "id": "httpbin-get",
    "methods": ["GET"],
    "uri": "/fakeget",
    "plugins": {
      "proxy-rewrite": {
        "uri": "/get",
        "use_real_request_uri_unsafe": true
      }
    },
    "upstream": {
      "type": "roundrobin",
      "pass_host": "node",
      "nodes": {
        "httpbin.org:80": 1
      }
    }
  }'
```
2. Send httpbin GET request with query param `curl -XGET "localhost:9080/fakeget?foo=bar"`

Image

3. Update proxy-rewrite plugin setting:
```
curl -XPUT -H "x-api-key:${ADMIN_KEY}" http://localhost:9180/admin/apisix/routes \
-d '{
    "id": "httpbin-get",
    "methods": ["GET"],
    "uri": "/fakeget",
    "plugins": {
      "proxy-rewrite": {
        "uri": "/get",
        "use_real_request_uri_unsafe": false
      }
    },
    "upstream": {
      "type": "roundrobin",
      "pass_host": "node",
      "nodes": {
        "httpbin.org:80": 1
      }
    }
  }'
```
4. Send the same request `curl -XGET "localhost:9080/fakeget?foo=bar"`

Image

### Environment

- APISIX version (run `apisix version`): 3.13.0
- Operating system (run `uname -a`): Debian GNU/Linux 11
- OpenResty / Nginx version (run `openresty -V` or `nginx -V`): openresty/1.27.1.2

### PR DESCRIPTION
**Pull request by YapWC**
…unsafe is true

When use_real_request_uri_unsafe is set to true, url that was requested with query string, arguments is not parsed. Expected to parse the arguments just like same default behavior when use_real_request_uri_unsafe is set to false.

The real issue lies in the fact that when use_real_request_uri_unsafe is set to true and conf.uri is provided, the core.utils.resolve_var output back an upstream uri that does not include query string. Therefore we need to concatenate the query string for the case if conf.uri is true.

#### Which issue(s) this PR fixes:

Fixes #12623 

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Review by Baoyuantop — COMMENTED** [review]
If the plugin only configures "use_real_request_uri_unsafe": true, it will cause the query to be concatenated repeatedly.

**Comment by YapWC** [comment]
Hi @Baoyuantop alright I think I understand what you mean. I think it is better for me to debug it line by line in this case. For your side normally how do you run debugging/printing code output?

File: apisix/plugins/proxy-rewrite.lua [hunk 1]
```diff
@@ -275,15 +275,27 @@ function _M.rewrite(conf, ctx)
         end
     end
 
+    -- normalized uri (without query string)
     local upstream_uri = ctx.var.uri
     local separator_escaped = false
+
+    local query_string = ""
+    -- When use_real_request_uri_unsafe is true, we use real_request_uri directly
```
Comment (Baoyuantop) [inline apisix/plugins/proxy-rewrite.lua]: Please simplify the comments.

**Comment by Baoyuantop** [comment]
Hi @YapWC, following up on the previous review comments. Please let us know if you have any updates. Thank you.

**Review by moonming — CHANGES_REQUESTED** [review]
Hi @YapWC, thank you for fixing the query string issue in proxy-rewrite!

Query strings being discarded when using `use_real_request_uri` is a real bug that affects production use. 

**Please**:
1. Rebase on the latest master
2. Add edge case tests: empty query strings, query strings with special characters, multiple `?` in the URI
3. Verify the fix works correctly with both `use_real_request_uri_unsafe` variants

Looking forward to the update!

**Comment by Baoyuantop** [comment]
Hi @YapWC, following up on the previous review comments. Please let us know if you have any updates. Thank you.

File: t/plugin/proxy-rewrite.t [hunk 2]
```diff
@@ -1231,3 +1231,95 @@ GET /hello
...
+        }
+    }
+--- request
+GET /t
+--- response_body
+passed
+
+
+
+=== TEST 46: rewrite uri args
```
Comment (nic-6443) [inline t/plugin/proxy-rewrite.t:1274]: The earlier revision of this PR double-appended the query when only `use_real_request_uri_unsafe: true` was set (no `uri`) — that is fixed now since `query_string` is only appended in the `conf.uri` branch, but none of the tests actually pin that scenario down. TEST 8/9 in proxy-rewrite3.t cover unsafe mode without a query string, and the new tests here all set `uri` together with the flag. Since this exact case already regressed once during review, it would be good to add a test with just `use_real_request_uri_unsafe: true` (no `uri`/`regex_uri`) and a request like `/hello?q=apisix`, asserting the query reaches the upstream exactly once.

**Review by shreemaan-abhishek — APPROVED** [review]

**Review by nic-6443 — APPROVED** [review]

**Review by membphis — APPROVED** [review]

**Review by moonming — APPROVED** [review]
